sh/README.adoc

```diff
@@ -0,0 +1,50 @@
+= Shell library docs
+:toc:
+:toclevels: 3
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+== Rough design plan
+
+CAUTION: Not yet fully implemented.
+
+Premise here is if you want the whole hog just source `sh/lib.sh` after setting `SOURCEPREFIX` to the directory containing
+that dir/file. For POSIX, shells there is no portable way to "know" where one imported from, so we just defer that to calling
+scripts as they can know where they're importing from.
+
+When you have that you get all the fun stuff.
+
+Goal(s):
+
+* Make debugging shell a bit easier and have some standard library of shell aiding
+* Promote defensive scripting; provide common error handling and helper functions to mitigate mistakes
+
+That will end up with something like scripts having:
+
+[source,sh]
+----
+SOURCEPREFIX=$CWD source ./sh/lib.sh
```

Comment:
I have to get out of the non portable source habit myself but since we're here lets be consistent:
```suggestion
SOURCEPREFIX=$(pwd)/sh
. ${SOURCEPREFIX}/lib.sh
```